binary_transparency/firmware/cmd/flash_tool/flash_tool.go

```diff
@@ -71,13 +73,20 @@ func main() {
 	// TODO(al): check signature on checkpoints when they're added.
 
 	var dev devices.Device
-	dev, err = dummy.NewFromFlags()
+	switch *deviceID {
+	case "armory":
+		dev, err = armory_flash.NewFromFlags()
+	case "dummy":
+		dev, err = dummy.NewFromFlags()
+	default:
+		glog.Exit("--device must be one of: 'dummy', 'armory'")
+	}
```

Comment:
BTW, https://golang.org/pkg/plugin/ would be a nifty solution to loading in these different devices. The flag business might be tricky actually, but worth considering if we want to truly have a generic flash tool.

Reply:
Yeah, good call - I'll merge this as is but go and investigate for a follow up PR.